redis/src/client.rs

```diff
@@ -287,6 +293,78 @@ impl Client {
     pub(crate) fn connection_info(&self) -> &ConnectionInfo {
         &self.connection_info
     }
+
+    /// Constructs a new `Client` with parameters necessary to create a TLS connection.
+    ///
+    /// - `conn_info` - URL using the `rediss://` scheme.
+    /// - `client_tls_params` - Optional pair of byte readers yielding PEM formatted files for:
+    ///   0. client's certificate chain
+    ///   1. client's private key
+    /// - `root_cert` - Optional byte readers yielding PEM formatted file for root certificates.
+    ///
+    /// If client cert+key pair is not provided, then client-side authentication is not enabled. If
+    /// trust anchors are not provided, then system root certificates are used instead.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::{fs::File, io::BufReader};
+    ///
+    /// use redis::{Client, AsyncCommands as _};
+    ///
+    /// async fn do_redis_code(
+    ///     url: &str,
+    ///     root_cert_file: &str,
+    ///     cert_file: &str,
+    ///     key_file: &str
+    /// ) -> redis::RedisResult<()> {
+    ///     let cert_file = File::open(cert_file).expect("cannot open private cert file");
+    ///     let mut reader_cert = BufReader::new(cert_file);
+    ///
+    ///     let key_file = File::open(key_file).expect("Cannot open private key file");
+    ///     let mut reader_key = BufReader::new(key_file);
+    ///
+    ///     let root_cert_file = File::open(root_cert_file).expect("Cannot open root cert file");
+    ///     let mut reader_root_certs = BufReader::new(root_cert_file);
+    ///
+    ///     let client = Client::build_with_tls(
+    ///         url,
+    ///         Some((&mut reader_cert, &mut reader_key)),
+    ///         Some(&mut reader_root_certs),
+    ///     )
+    ///     .expect("Unable to build client");
+    ///
+    ///     let connection_info = client.get_connection_info();
+    ///
+    ///     println!(">>> connection info: {connection_info:?}");
+    ///
+    ///     let mut con = client.get_async_connection().await?;
+    ///
+    ///     con.set("key1", b"foo").await?;
+    ///
+    ///     redis::cmd("SET")
+    ///         .arg(&["key2", "bar"])
+    ///         .query_async(&mut con)
+    ///         .await?;
+    ///
+    ///     let result = redis::cmd("MGET")
+    ///         .arg(&["key1", "key2"])
+    ///         .query_async(&mut con)
+    ///         .await;
+    ///     assert_eq!(result, Ok(("foo".to_string(), b"bar".to_vec())));
+    ///     println!("Result from MGET: {result:?}");
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
+    #[cfg(feature = "tls-rustls")]
+    pub fn build_with_tls<C: IntoConnectionInfo>(
+        conn_info: C,
+        client_tls_params: Option<(&mut dyn BufRead, &mut dyn BufRead)>,
```

Comment:
Minor, but I wonder if we should add a type here so it's a bit more obvious which is the key and which is the cert. (same comment for cluster_client)

Reply:
Was thinking about this and I'm feeling tempted to make these structs public and using `RawCertificates` as the set TLS parameters for both clustered and single client constructor. 
```
pub struct ClientTlsBinary {
    pub client_cert: Vec<u8>,
    pub client_key: Vec<u8>,
}

pub struct CertificatesBinary {
    pub client_tls: Option<ClientTlsBinary>,
    pub root_cert: Option<Vec<u8>>,
}
```
Wdyt?

Reply:
Looks good to me :+1:

Reply:
cool. done (hopefully final form now :crossed_fingers: )